deluge/ui/gtkui/gtkui.py

```diff
@@ -238,36 +239,52 @@ def nsapp_open_file(osxapp, filename):
         self.rpc_stats = LoopingCall(self.print_rpc_stats)
 
         # Twisted catches signals to terminate, so have it call a pre_shutdown method.
-        reactor.addSystemEventTrigger("before", "shutdown", self.pre_shutdown)
+        reactor.addSystemEventTrigger("before", "gtkui_close", self.close)
+
         reactor.callWhenRunning(self._on_reactor_start)
+        self.closing = False
+
+        def gtkui_sigint_handler(num, frame):
+            log.debug("SIGINT signal caught - firing event: 'gtkui_close'")
+            reactor.callLater(0, reactor.fireSystemEvent, 'gtkui_close')
+
+        signal.signal(signal.SIGINT, gtkui_sigint_handler)
```

Comment:
To me this seems like the exact same as if twisted catches the sigint before shutdown.

Why not simply `reactor.addSystemEventTrigger("before", "shutdown", self.close)`?